### ISSUE TITLE
Retrieving more than 20 members (Thru paged results) not possible

### ISSUE THREAD

**Issue by DurtyFree** [sawch/Patreon.Net#1]
Hi there,

first I want to thank you for sharing this great Patreon.NET library - finally someone did something really useable!
From my first quick tests, everything besides retrieving the members with all pages, work great.

On my patreon campaign there are currently almost 800 members, the API tho returns me only the first 20 of those. I think this is related to the pages not being called one after the other.
Maybe add some parameter to decide whether a given page should be retrieved only - or all pages?

Best regards
Durty

**Comment by DurtyFree** [comment]
After digging a bit into the code, I found the method to retrieve members for a certain page.

I created some API for myself to just retrieve all existing pages, sharing this for others here (Put into `PatreonClient.cs`):
```csharp
/// <summary>
/// Gets all members for a given campaign owned by the authorized user.
/// </summary>
/// <param name="campaignId">The ID of the campaign to fetch the members of.</param>
/// <param name="includes">The desired resources to be included on the <see cref="Member"/> objects in the returned array.</param>
/// <returns>A resource array of members, or <see langword="null"/> if no campaign with the given <paramref name="campaignId"/> is found.</returns>
/// <exception cref="PatreonApiException"/>
public async Task<ResourceArray<Member, Member.Relationships>> GetAllCampaignMembersAsync(string campaignId, Includes includes = Includes.None)
{
    if (string.IsNullOrWhiteSpace(campaignId))
        throw new ArgumentException("Value cannot be null, empty or whitespace.", nameof(campaignId));

    var members = await GetCampaignMembersAsync(campaignId, Includes.All).ConfigureAwait(false);
    if (members.Meta.Pagination.Total < 20)
        return members;
    
    List<ResourceData<Member, Member.Relationships>> includingPagedMembers = members.Data.ToList();
    do
    {
        members = await GetCampaignMembersAsync(campaignId, members.Meta.Pagination.Cursor.Next, Includes.All).ConfigureAwait(false);
        includingPagedMembers.AddRange(members.Data);
    } while (!string.IsNullOrEmpty(members.Meta.Pagination?.Cursor?.Next));

    members.Meta.Pagination.Cursor = null;
    members.Data = includingPagedMembers.ToArray();
    return members;
}
```

@sawch Feel free to add this or something similar :D

**Comment by DurtyFree** [comment]
Also the note on the pagination, in the README, can be removed. I tested it with more than 700 members and it works flawless. Thanks.

**Comment by sawch** [comment]
Nice! That's great to hear. I agree, it would be preferable to have the library handle pages automatically - I think an enumerable that fetches the next page if it encounters one when iterating would be preferable to allocating an additional list and array. I'll have a jab at implementing that when I get some free time, thank you for confirming that it works!

**Comment by sawch** [comment]
@DurtyFree I have a quick question for documentation - is the `Pagination.Total` the total number of pages, total number of results in the current page or total results altogether? I only have 1 member so I can't tell.

**Comment by DurtyFree** [comment]
> 
> 
> @DurtyFree I have a quick question for documentation - is the `Pagination.Total` the total number of pages, total number of results in the current page or total results altogether? I only have 1 member so I can't tell.

Actually Pagination.Total is the total amount of members you have (Counting all, active patrons, former patrons etc.). To get the total amount of pages you need to divide it by 20, as this is what it seems to return with every page.

**Comment by sawch** [comment]
I see, thank you! I merged my changes in so you can give them a peek if you want. I'll wait a bit before closing these issues and going forward in case you do check them out and there's still any problems.

**Comment by DurtyFree** [comment]
I have tested all those changes and it seems to work very well, I hope to see this as the final addition with a nuget release soon:)

**Comment by DurtyFree** [comment]
I have also tested the implementation for
`await foreach(var member in members)`
which works without issues for me (Tested with more than 800 members returning)

**Comment by sawch** [comment]
Awesome! Yep I'll be doing that soon.

**Comment by sawch** [comment]
Fixed with #5!
